runbotics/runbotics-desktop/src/action/variable/variable.action-handler.ts

```diff
@@ -62,17 +62,20 @@ export default class VariableActionHandler extends StatelessActionHandler {
         return actionInput;
     }
 
-    private async fetchGlobalVariable(globalVariableId: number): Promise<AssignVariableActionInput> {
-        const response: AxiosResponse<IGlobalVariable> = await orchestratorAxios.get(`/api/global-variables/${globalVariableId}`);
-        return this.mapGlobalVariableToActionInput(response.data);
+    private fetchGlobalVariable(globalVariableId: number): Promise<AxiosResponse<IGlobalVariable>> {
+        return orchestratorAxios.get(`/api/global-variables/${globalVariableId}`);
     }
 
     private async assignGlobalVariable (
         request: DesktopRunRequest<'variables.assignGlobalVariable', AssignGlobalVariableActionInput>
     ) {
+        const userId = request.userId;
         const globalVariableIds = request.input.globalVariables;
         for (const globalVariableId of globalVariableIds) {
-            const input = await this.fetchGlobalVariable(globalVariableId);
+            const { data: globalVariable } = await this.fetchGlobalVariable(globalVariableId);
+            if (globalVariable.creator.id !== userId)
+                throw new Error('One of global variables does not belong to the user');
+            const input = this.mapGlobalVariableToActionInput(globalVariable);
```

Comment:
dont use `userId`. As we talked recently, user info is already in the token. Which user started a process should not matter here. Throw a custom error whenever `fetchGlobalVariable` has thrown